zerver/openapi/zulip.yaml

```diff
@@ -2748,7 +2748,40 @@ paths:
                                 Event sent to a user when [message flags][message-flags] are added
                                 to messages.
 
+                                This can reflect a direct user action, or can be the indirect
+                                consequence of another action. Whatever the cause, if there's a change
+                                in the set of message flags that the user has for a message, then an
+                                `update_message_flags op: add` event will be sent with the change. Note
+                                that this applies when the user already had access to the message, and
+                                continues to have access to it. When a message newly appears or
+                                disappears, a [`message`][message-event] or
+                                [`delete_message`][message-delete] event is sent instead.
+
+                                Some examples of actions that trigger an `update_message_flags op: add`
+                                event:
+
+                                - The `"starred"` flag is added when the user chooses to [star a
+                                  message](/help/star-a-message).
+                                - The `"read"` flag is added when the user marks messages as read by
+                                  scrolling through them, or uses [Mark all messages as read][all-read]
+                                  on a conversation.
+                                - The `"read"` flag is added when the user [mutes](/help/mute-a-user) a
+                                  message's sender.
+                                - The `"read"` flag is added after the user unsubscribes from a stream,
+                                  or messages are moved to a not-subscribed stream, provided the user
+                                  can still access the messages at all.
+
+                                In some cases, a change in message flags that's caused by another change
+                                may happen a short while after the original change, rather than
+                                simultaneously. For example, when messages that were unread are moved to
+                                a stream where the user is not subscribed, the resulting change in
+                                message flags (and the corresponding `update_message_flags op: add`
+                                event with flag `"read"`) may happen later than the message move itself.
```

Comment:
Maybe clearest would be something like this:

"While in most cases, events for changes in message flags resulting from other changes will be sent at the same time as the changes themselves (within hundreds of milliseconds), in some code paths where expensive database operations are involved, a deferred work queue processor is used, which can in rare cases backlog, resulting in a larger delay. For example, ... ('when messages are moved ... with flag `"read"`' goes here) ... uses such a queue processor and thus clients may receive events somewhat than they receive the message move event itself."

Reply:
Hmm, "within hundreds of milliseconds" is not what I'd call simultaneous — from a client's perspective, that is a short while later. In particular it means that the client will be showing the user a UI, for long enough for the user to see it, based on having received the one change but not yet the other.

What are other examples where you'd expect that kind of delay?